### PR TITLE
Add Pearson correlation aggregator to Scala

### PR DESCRIPTION
**Pull request by tpoterba**
PRs to follow will add to Python, and add array correlation function.

### CODE REVIEW

File: hail/src/main/scala/is/hail/annotations/aggregators/RegionValuePearsonCorrelationAggregator.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package is.hail.annotations.aggregators
+
+import is.hail.annotations.{Region, RegionValueBuilder}
+import is.hail.stats.PearsonCorrelationCombiner
+
+class RegionValuePearsonCorrelationAggregator extends RegionValueAggregator {
+  var combiner = new PearsonCorrelationCombiner()
+
+  def seqOp(region: Region, x: Double, xm: Boolean, y: Double, ym: Boolean) {
+    if (!xm && !xm)
```
Comment (jigold) [inline hail/src/main/scala/is/hail/annotations/aggregators/RegionValuePearsonCorrelationAggregator.scala]: This should be `!xm && !ym`

Comment (tpoterba) [reply hail/src/main/scala/is/hail/annotations/aggregators/RegionValuePearsonCorrelationAggregator.scala]: thanks! Will make sure this is tested too

Comment (tpoterba) [reply hail/src/main/scala/is/hail/annotations/aggregators/RegionValuePearsonCorrelationAggregator.scala]: fixed. verified that my test caught this.

File: hail/src/main/scala/is/hail/stats/PearsonCorrelationCombiner.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  def merge(other: PearsonCorrelationCombiner) {
+    xSum += other.xSum
+    ySum += other.ySum
+    xSqSum += other.xSqSum
+    ySqSum += other.ySqSum
+    xySum += other.xySum
+    n += other.n
+  }
+
+  private def computeResult(): Double = (n * xySum - xSum * ySum) / (math.sqrt(n * xSqSum - xSum * xSum) * math.sqrt(n * ySqSum - ySum * ySum))
```
Comment (jbloom22) [inline hail/src/main/scala/is/hail/stats/PearsonCorrelationCombiner.scala]: replace denominator with `math.sqrt((n * xSqSum - xSum * xSum) * (n * ySqSum - ySum * ySum))` to save a sqrt

Comment (tpoterba) [reply hail/src/main/scala/is/hail/stats/PearsonCorrelationCombiner.scala]: done

**Pull request merged**